test/.eslintrc.yaml

```diff
@@ -8,47 +8,47 @@ rules:
-      message: "Use an object as second argument of `assert.throws()`."
-    - selector: "CallExpression:matches([callee.name='throws'], [callee.property.name='throws'])[arguments.length<2]"
-      message: "`assert.throws()` must be invoked with at least two arguments."
-    - selector: "CallExpression[callee.name='setTimeout'][arguments.length<2]"
-      message: "`setTimeout()` must be invoked with at least two arguments."
-    - selector: "CallExpression[callee.name='setInterval'][arguments.length<2]"
-      message: "`setInterval()` must be invoked with at least two arguments."
-    - selector: "ThrowStatement > CallExpression[callee.name=/Error$/]"
-      message: "Use `new` keyword when throwing an `Error`."
-    - selector: "CallExpression:matches([callee.name='notDeepStrictEqual'], [callee.property.name='notDeepStrictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
-      message: "The first argument should be the `actual`, not the `expected` value."
-    - selector: "CallExpression:matches([callee.name='notStrictEqual'], [callee.property.name='notStrictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
-      message: "The first argument should be the `actual`, not the `expected` value."
-    - selector: "CallExpression:matches([callee.name='deepStrictEqual'], [callee.property.name='deepStrictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
-      message: "The first argument should be the `actual`, not the `expected` value."
-    - selector: "CallExpression:matches([callee.name='strictEqual'], [callee.property.name='strictEqual'])[arguments.0.type='Literal']:not([arguments.1.type='Literal']):not([arguments.1.type='ObjectExpression']):not([arguments.1.type='ArrayExpression']):not([arguments.1.type='UnaryExpression'])"
-      message: "The first argument should be the `actual`, not the `expected` value."
-    - selector: "CallExpression[callee.name='isNaN']"
-      message: "Use Number.isNaN() instead of the global isNaN() function."
-    - selector: "VariableDeclarator > CallExpression:matches([callee.name='debuglog'], [callee.property.name='debuglog']):not([arguments.0.value='test'])"
-      message: "Use 'test' as debuglog value in tests."
+    - selector: CallExpression:matches([callee.name='deepStrictEqual'], [callee.property.name='deepStrictEqual']):matches([arguments.1.type='Literal']:not([arguments.1.regex]), [arguments.1.type='Identifier'][arguments.1.name='undefined'])
```

Comment:
I don't feel comfortable seeing long unquoted strings like that which contain special characters (including quotes!).
Isn't it less mental overhead to generally favor quotes?

Reply:
Personally, I find the reading clearer when there are no quotation marks, but I think that's just because I'm used to not putting them.

Reply:
It's not really about reading it, but about knowing whether it will effectively be parsed as a string.

Reply:
Yes it will always be parsed as a string. And anyway if there is a problem, the linter would fail.